### PR TITLE
Feat/m foundation central

### PR DESCRIPTION
**Pull request by abhimutant**

### CODE REVIEW

File: client/client.go [hunk 1]
```diff
@@ -350,8 +350,9 @@ func (c *Client) Do(ctx context.Context, req *http.Request, v interface{}) error
 	}
 
 	req = req.WithContext(ctx)
+	fmt.Printf("req: %v\n", req)
```
Comment (siddharth-kulshrestha) [inline client/client.go]: Is it required? 
If yes then use log.

Comment (siddharth-kulshrestha) [reply client/client.go]: Make changes everywhere.

File: client/fc/fc.go [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+	}
+
+	fc := &Client{
+		client: c,
+		Service: Operations{
+			client: c,
+		},
+	}
+
+	// f.client.OnRequestCompleted(func(req *http.Request, resp *http.Response, v interface{}) {
```
Comment (siddharth-kulshrestha) [inline client/fc/fc.go]: Do we need this commented code?

File: client/fc/fc_service.go [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	"github.com/terraform-providers/terraform-provider-nutanix/client"
+)
+
+// Operations ...
```
Comment (siddharth-kulshrestha) [inline client/fc/fc_service.go]: Please use proper comments. 
Make changes everywhere.

File: nutanix/config.go [hunk 4]
```diff
@@ -55,11 +56,16 @@ func (c *Config) Client() (*Client, error) {
...
+	fcClient, err := fc.NewFoundationCentralClient(configCreds)
+	if err != nil {
+		return nil, err
+	}
 	return &Client{
 		WaitTimeout:         c.WaitTimeout,
 		API:                 v3Client,
 		KarbonAPI:           karbonClient,
 		FoundationClientAPI: foundationClient,
+		FC:                  fcClient,
```
Comment (siddharth-kulshrestha) [inline nutanix/config.go]:
```suggestion
		FoundationCentral:                  fcClient,
```
Please use full name here

File: client/fc/fc_structs.go [hunk 5]
```diff
@@ -0,0 +1,312 @@
+package fc
```
Comment (siddharth-kulshrestha) [inline client/fc/fc_structs.go]: can we rename this package to foundation_central ?

File: client/fc/fc_structs.go [hunk 6]
```diff
@@ -0,0 +1,312 @@
...
+	NodeSerial         *string            `json:"node_serial,omitempty"`
+	BlockSerial        *string            `json:"block_serial,omitempty"`
+	HypervisorType     *string            `json:"hypervisor_type,omitempty"`
+	HypervisorNetmask  *string            `json:"hypervisor_netmask,omitempty"`
+	HypervisorGateway  *string            `json:"hypervisor_gateway,omitempty"`
+	Model              *string            `json:"model,omitempty"`
+	AosVersion         *string            `json:"aos_version,omitempty"`
+}
+
+// ImagedNodeResponse ...
```
Comment (siddharth-kulshrestha) [inline client/fc/fc_structs.go:106]: Please add comments everywhere.

**Review by siddharth-kulshrestha — CHANGES_REQUESTED** [review]

**Review by siddharth-nutanix — APPROVED** [review]

**Review by siddharth-kulshrestha — APPROVED** [review]

**Pull request merged**